script_test.go

```diff
@@ -1286,6 +1286,39 @@ func TestFindFiles_InNonexistentPathReturnsError(t *testing.T) {
 	}
 }
 
+func TestFindFiles_DoesNotErrorWhenSubDirectoryIsNotReadable(t *testing.T) {
+	t.Parallel()
+	tmpDir := t.TempDir()
+	fileAPath := filepath.Join(tmpDir, "file_a.txt")
+	if err := os.WriteFile(fileAPath, []byte("hello world!"), os.ModePerm); err != nil {
+		t.Fatal(err)
+	}
+	restrictedDirPath := filepath.Join(tmpDir, "restricted_dir")
+	if err := os.Mkdir(restrictedDirPath, os.ModePerm); err != nil {
+		t.Fatal(err)
+	}
+	fileBPath := filepath.Join(restrictedDirPath, "file_b.txt")
+	if err := os.WriteFile(fileBPath, []byte("hello again!"), os.ModePerm); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.Chmod(restrictedDirPath, 0o000); err != nil {
```

Comment:
@bitfield this test fails on windows since chmod is a unix command. thoughts on how to proceed? should we make this a unix-only test?

Reply:
Yes, Windows doesn't have the concept of file permissions as Go understands them—this is Unix-only.